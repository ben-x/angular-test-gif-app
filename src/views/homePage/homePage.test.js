import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../utils/index";
import HomePage from "./HomePage";

const setUp = (props = {}) => {
    const component = shallow(<HomePage {...props} />);
    return component;
};

describe("Article Banner", () => {
    describe("Child components should render", () => {
        let component;
        beforeEach(() => {
            component = setUp();
        });
        it("Should render top searchbar on home page", () => {
            const wrapper = findByTestAttr(component, "searchbar");
            expect(wrapper.length).toBe(1);
        });
        it("Should render top searchbar on home page", () => {
            const wrapper = findByTestAttr(component, "gifdetails");
            expect(wrapper.length).toBe(0);
        });
        it("Should render top searchbar on home page", () => {
            const wrapper = findByTestAttr(component, "gifbox");
            expect(wrapper.length).toBe(1);
        });
        it("Should render successfully", () => {
            expect(component).toMatchSnapshot();
        });
    });
});
