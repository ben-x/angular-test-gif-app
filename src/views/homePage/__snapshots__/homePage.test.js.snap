// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Banner Child components should render Should render successfully 1`] = `ShallowWrapper {}`;
